### PR TITLE
[DSD-5159] Added helm for mosipcertmanager

### PR DESCRIPTION
**Pull request by VSIVAKALYAN**

### CODE REVIEW

**Review by syedsalman3753 — CHANGES_REQUESTED** [review]
please change the chart name to `mosipcertmanager`

File: charts/certvalidator/values.yaml [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+    #     for: 5m
+    #     labels:
+    #       severity: error
+    rules: []
+
+## Admin swagger should have only internal access. Hence linked to internal gateway
+
+types:
+  - name: full
+    enabled: true
```
[the remark this answers is not in the document]
Comment (VSIVAKALYAN) [reply charts/certvalidator/values.yaml]: removed & hence closing Conversation

File: charts/certvalidator/templates/cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
+{{- range $type := $.Values.types }}
+{{- if $type.enabled }}
```
[the remark this answers is not in the document]
Comment (VSIVAKALYAN) [reply charts/certvalidator/templates/cronjob.yaml]: Removed & hence closing this Conversation.

File: charts/certvalidator/templates/cronjob.yaml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                {{- range $.Values.extraEnvVarsSecret }}
+                  - secretRef:
+                        name: {{ . }}
+                {{- end }}
+                {{- end }}
+            ports:
+                - name: spring-service
+                  containerPort: {{ $.Values.springServicePort }}
+{{- end }}
+{{- end }}
```
[the remark this answers is not in the document]
Comment (VSIVAKALYAN) [reply charts/certvalidator/templates/cronjob.yaml]: Removed & hence closing this Conversation.

File: charts/certvalidator/values.yaml [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+
+certvalidator:
+  configmaps:
+    db:
+      db_port: '5432'
+      db_suuser: 'postgres'
+      db_server: 'api-internal.sandbox.xyz.net'
+    preexpirydays: '7'
+  secrets:
+    mosip-pms-client-secret:
```
[the remark this answers is not in the document]
Comment (VSIVAKALYAN) [reply charts/certvalidator/values.yaml]: Removed & hence closing this Conversation.

File: charts/certvalidator/values.yaml [hunk 5]
```diff
@@ -0,0 +1,437 @@
...
+    enabled: true
+
+crontime: "0 3 * * *"    ## run cronjob every day at 3 AM (time hr: 0-23 )
+
+certvalidator:
+  configmaps:
+    db:
+      db_port: '5432'
+      db_suuser: 'postgres'
+      db_server: 'api-internal.sandbox.xyz.net'
```
[the remark this answers is not in the document]
Comment (VSIVAKALYAN) [reply charts/certvalidator/values.yaml]: Removed & hence closing this Conversation.

File: charts/certvalidator/values.yaml [hunk 6]
```diff
@@ -0,0 +1,437 @@
...
+    preexpirydays: '7'
```
[the remark this answers is not in the document]
Comment (VSIVAKALYAN) [reply charts/certvalidator/values.yaml]: Removed & hence closing this Conversation.

File: charts/certvalidator/values.yaml [hunk 7]
```diff
@@ -0,0 +1,437 @@
...
+##     value: "bar"
+##
+extraEnvVars: []
+
+## ConfigMap with extra environment variables
+##
+extraEnvVarsCM:
+  - global
+  - db
+  - pre-expirydays
```
[the remark this answers is not in the document]
Comment (VSIVAKALYAN) [reply charts/certvalidator/values.yaml]: taken care & hence closing this Conversation.

File: charts/certvalidator/values.yaml [hunk 8]
```diff
@@ -0,0 +1,437 @@
...
+## Secret with extra environment variables
+##
+extraEnvVarsSecret:
+  - s3
+  - postgres-postgresql
+  - mosip-pms-client-secret
```
[the remark this answers is not in the document]
Comment (VSIVAKALYAN) [reply charts/certvalidator/values.yaml]: taken care & hence closing this Conversation.

**Comment by VSIVAKALYAN** [comment]
Connected with @Mahesh-Binayak tested the Deployment in dev-staging env 
Everything Working Fine pls approve & merge this PR.

File: charts/mosipcertmanager/Chart.yaml [hunk 9]
```diff
@@ -0,0 +1,19 @@
+apiVersion: v2
+name: mosipcertmanager
+description: A Helm chart to deploy mosipcertmanager to test working of MOSIP modules
+type: application
+version: 12.0.2
```
Comment (syedsalman3753) [inline charts/mosipcertmanager/Chart.yaml:5]: please change the version to 0.0.1-develop

**Review by syedsalman3753 — CHANGES_REQUESTED** [review]

File: charts/mosipcertmanager/values.yaml [hunk 10]
```diff
@@ -0,0 +1,433 @@
...
+
+crontime: "0 3 * * *"    ## run cronjob every day at 3 AM (time hr: 0-23 )
+
+mosipcertmanager:
+  configmaps:
+    db:
+      db-port: '5432'
+      db-su-user: 'postgres'
+      db-server: 'postgres-k8s-svc-name'
+      db-host: 'mosip-api-internal-host'
```
Comment (syedsalman3753) [inline charts/mosipcertmanager/values.yaml:429]: please check developer which env variable will be used by mosipcertmanager

and hardcode the postgres k8s service name

**Review by syedsalman3753 — CHANGES_REQUESTED** [review]

File: charts/mosipcertmanager/values.yaml (hunk 10 above)
Comment (VSIVAKALYAN) [reply charts/mosipcertmanager/values.yaml:429]: db-host is required & hence removed db-sever & deployed & tested in dev-staging env & Job is Working as expected.

**Review by syedsalman3753 — APPROVED** [review]

**Review by ckm007 — APPROVED** [review]

**Pull request merged**